File: php/test/Creational/PrototypeTest.php
<?

require_once 'src/Creational/Prototype/SoupSpoon.php';
require_once 'src/Creational/Prototype/SaladFork.php';
require_once 'src/Creational/Prototype/SaladSpoon.php';
require_once 'src/Creational/Prototype/PrototypeFactory.php';

class Creational_PrototypeTest extends PHPUnit_Framework_TestCase
{

	function test()
	{
		$prototypeFactory = new PrototypeFactory(
			new SoupSpoon(), 
			new SaladFork());
		$spoon = $prototypeFactory->makeSpoon();
		$fork = $prototypeFactory->makeFork();
		$this->assertEquals("Spoon: " . $spoon->spoonName . ", Fork: " . $fork->forkName ,'Spoon: Soup Spoon, Fork: Salad Fork', 'Testing with SoupSpoon and SaladFork');

		$prototypeFactory = new PrototypeFactory(
			new SaladSpoon(), 
			new SaladFork());
		$spoon = $prototypeFactory->makeSpoon();
		$fork = $prototypeFactory->makeFork();
		$this->assertEquals("Spoon: " . $spoon->spoonName . 
			", Fork: " . $fork->forkName,'Spoon: Salad Spoon, Fork: Salad Fork', 'Testing with SaladSpoon and SaladFork');
	}
}

File: go/creational/prototype/prototype_test.go
//TestPrototype - testing the Prototype

package prototype

import (
	"testing"
	"../../lib/asserts"
)

func TestPrototype (t *testing.T ) { 

	prototypeFactory := PrototypeFactory{NewSoupSpoon(), NewSaladFork()}
	spoon := prototypeFactory.MakeSpoon()
	fork := prototypeFactory.MakeFork()
	asserts.Equals( t, "Getting the Spoon and Fork name:", 
		"Spoon: Soup Spoon, Fork: Salad Fork", 
		"Spoon: " + spoon.GetName() + ", Fork: " + fork.GetName())


	prototypeFactory = PrototypeFactory{NewSaladSpoon(), NewSaladFork()}
	spoon = prototypeFactory.MakeSpoon()
	fork = prototypeFactory.MakeFork()
	asserts.Equals( t, "Creating a Prototype Factory with a SaladSpoon and a SaladFork", 
		"Spoon: Salad Spoon, Fork: Salad Fork",
		"Spoon: " + spoon.GetName() + ", Fork: " + fork.GetName() )

}      
